### PR TITLE
Autogen modules helper

### LINKED ISSUE #525: Add generate_modules helper
It’s a common situation when you'd want to generate a module from some template files, e.g. https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/helpers/generate_modules/generate_modules.py so it makes sense to put the helper inside the devtools image and not populate it in each module.

### PR DESCRIPTION
**Pull request by omazin**
Fixes #525

### CODE REVIEW

**Comment by morgante** [comment]
@omazin Before we can merge this, can you please provide a PR showing how this would also work for the GKE module?

**Comment by omazin** [comment]
> @omazin Before we can merge this, can you please provide a PR showing how this would also work for the GKE module?

@morgante I've already created one for [GKE](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/389) and another one [VM](https://github.com/terraform-google-modules/terraform-google-vm/pull/59) modules.

**Comment by morgante** [comment]
@omazin Thanks. Please make sure https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/378 could also be accommodated.

File: infra/build/developer-tools/build/scripts/task_wrapper_scripts/generate_modules.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        )
+        templates = env.list_templates()
+
+        for template_file in templates:
+            template = env.get_template(template_file)
+            if template_file.endswith(".tf.tmpl"):
+                template_file = template_file.replace(".tf.tmpl", ".tf")
+            rendered = template.render(
+                module.template_options(
+                    {'autogeneration_note': AUTOGEN_NOTE + template_folder}
```
Comment (morgante) [inline infra/build/developer-tools/build/scripts/task_wrapper_scripts/generate_modules.py]: Please make this into a format string, not a concat (cf. https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/helpers/generate_modules/generate_modules.py#L25)

**Comment by omazin** [comment]
> Also please make sure to add jinja and other requirements to the [image](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/build/developer-tools/build/data/requirements.txt).

@morgante, thought if jinja is installed here https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/build/developer-tools/build/install_dependencies.sh#L76 then we don't need to specify it as a requirement.

File: infra/build/developer-tools/build/scripts/task_wrapper_scripts/generate_modules.py (hunk 1 above)
Comment (omazin) [reply infra/build/developer-tools/build/scripts/task_wrapper_scripts/generate_modules.py]: Fixed

**Pull request merged**